### PR TITLE
[Keyvault] `az keyvault key encrypt/decrypt`: Support AES algorithm for MHSM

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

Feature asked by #19600 
`az keyvault key encrypt/decrypt`:
- migrate to track2
- support AES algorithm

Also refined some track1 code for easier understanding

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Keyvault

**Review by calvinhzy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py [hunk 1]
```diff
@@ -1282,7 +1282,7 @@ def test_keyvault_hsm_key_random(self):
         self.assertIsNotNone(result['value'])
 
     # Since the MHSM has to be activated manually so we use fixed hsm resource and mark the test as record_only
-    #@record_only()
+    @record_only()
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py]: We may remove the comment now.

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 2]
```diff
@@ -207,6 +207,13 @@
     the key. This operation is supported for asymmetric keys as a convenience for callers that
     have a key-reference but do not have access to the public key material. This operation
     requires the keys/encrypt permission.
+examples:
+  - name: Encrypt value(Base64 encoded string) with valut's key using RSA-OAEP.
+    text: |
+        az keyvault key encrypt -n mykey --vault-name myvault -a RSA-OAEP --value "YWJjZGVm" --data-type base64
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:213]: A best practice is to use **full argument name** in examples for clarity.

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 3]
```diff
@@ -1123,11 +1123,11 @@ def delete_policy(cmd, client, resource_group_name, vault_name,
 
 
 # region KeyVault Key
-def create_key(client, key_name=None, protection=None,  # pylint: disable=unused-argument
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py:1126]: The argument registrations of `key_name` and `name` seem duplicated:

For `key_name`:

https://github.com/Azure/azure-cli/blob/2af2d4f78d9d5867d7d889c0a47fe0da21b944b7/src/azure-cli/azure/cli/command_modules/keyvault/_params.py#L276-L279

For `name`:

https://github.com/Azure/azure-cli/blob/2af2d4f78d9d5867d7d889c0a47fe0da21b944b7/src/azure-cli/azure/cli/command_modules/keyvault/_params.py#L452-L456

I think we should remove one.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py:1126]: Exactly. That's what I did in `_params.py`. I have removed the track1 definition `key_name`.

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py (hunk 2 above)
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:213]:
```suggestion
        az keyvault key encrypt --name mykey --vault-name myvault --algorithm RSA-OAEP --value "YWJjZGVm" --data-type base64
```

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 4]
```diff
@@ -207,6 +207,13 @@
...
+  - name: Encrypt value(plaintext) with MHSM's key using AES-GCM.
+    text: |
+        az keyvault key encrypt -n mykey --hsm-name myhsm -a A256GCM --value "this is plaintext" --data-type plaintext --aad "101112131415161718191a1b1c1d1e1f"
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:216]:
```suggestion
        az keyvault key encrypt --name mykey --hsm-name myhsm --algorithm A256GCM --value "this is plaintext" --data-type plaintext --aad "101112131415161718191a1b1c1d1e1f"
```

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 5]
```diff
@@ -218,6 +225,14 @@
     and the algorithm to be used. The DECRYPT operation applies to asymmetric and symmetric keys
     stored in Vault or HSM since it uses the private portion of the key. This operation
     requires the keys/decrypt permission.
+examples:
+  - name: Decrypt value(Base64 encoded string returned by encrypt command) with valut's key using RSA-OAEP and get result as base64 encoded.
+    text: |
+        az keyvault key decrypt -n mykey --vault-name myvault -a RSA-OAEP --data-type base64 --value "CbFcCxHG7WTU+nbpFRrHoqSduwlPy8xpWxf1JxZ2y12BY/qFJirMSYq1i4SO9rvSmvmEMxFV5kw5s9Tc+YoKmv8X6oe+xXx+JytYV8obA5l3OQD9epuuQHWW0kir/mp88lzhcYWxYuF7mKDpPKDV4if+wnAZqQ4woB6t2JEZU5MVK3s+3E/EU4ehb5XrVxAl6xpYy8VYbyF33uJ5s+aUsYIrsVtXgrW99HQ3ic7tJtIOGuWqKhPCdQRezRkOcyxkJcmnDHOLjWA/9strzzx/dyg/t884gT7qrkmIHh8if9SFal/vi1h4XhoDqUleMTnKev2IFHyDNcYVYG3pftJiuA=="
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:231]:
```suggestion
        az keyvault key decrypt --name mykey --vault-name myvault --algorithm RSA-OAEP --data-type base64 --value "CbFcCxHG7WTU+nbpFRrHoqSduwlPy8xpWxf1JxZ2y12BY/qFJirMSYq1i4SO9rvSmvmEMxFV5kw5s9Tc+YoKmv8X6oe+xXx+JytYV8obA5l3OQD9epuuQHWW0kir/mp88lzhcYWxYuF7mKDpPKDV4if+wnAZqQ4woB6t2JEZU5MVK3s+3E/EU4ehb5XrVxAl6xpYy8VYbyF33uJ5s+aUsYIrsVtXgrW99HQ3ic7tJtIOGuWqKhPCdQRezRkOcyxkJcmnDHOLjWA/9strzzx/dyg/t884gT7qrkmIHh8if9SFal/vi1h4XhoDqUleMTnKev2IFHyDNcYVYG3pftJiuA=="
```

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 6]
```diff
@@ -218,6 +225,14 @@
...
+  - name: Decrypt value(Base64 encoded string returned by encrypt command) with MHSM's key using AES-GCM and get result as plaintext.
+    text: |
+        az keyvault key decrypt -n mykey --hsm-name myhsm -a A256GCM --value "N5w02jS77xg536Ddzv/xPWQ=" --data-type plaintext
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:234]:
```suggestion
        az keyvault key decrypt --name mykey --hsm-name myhsm --algorithm A256GCM --value "N5w02jS77xg536Ddzv/xPWQ=" --data-type plaintext
```

**Review by jiasli — APPROVED** [review]

**Pull request merged**